addons/l10n_tr_nilvera_einvoice/models/account_edi_xml_ubl_tr.py

```diff
@@ -42,8 +44,19 @@ def _get_formatted_id(invoice):
             'pricing_currency_code': invoice.currency_id.name.upper() if invoice.currency_id != invoice.company_id.currency_id else False,
             'currency_dp': 2,
         })
+        vals['vals']['note_vals'].append({'note': self._get_amount_in_text(invoice.amount_residual_signed, self.env.ref('base.TRY')), 'note_attrs': {}})
+        if vals['invoice'].currency_id.name != 'TRY':
+            vals['vals']['note_vals'].append({'note': self._get_amount_in_text(invoice.amount_residual, vals['invoice'].currency_id), 'note_attrs': {}})
         return vals
 
+    def _get_amount_in_text(self, amount, currency):
+        amount_i, amount_d = divmod(amount, 1)
```

Comment:
Beware that the amount might be negative and that divmod probably doesn't do what you want in that case.
If you want to keep the sign, you can use math.modf.
`divmod(-4.6, 1) => (-5.0, 0.40000000000000036)`